examples/CMakeLists.txt

```diff
@@ -20,3 +20,13 @@ add_executable(client_test
     torrent_view.cpp
     session_view.cpp)
 target_link_libraries(client_test PRIVATE torrent-rasterbar)
+
+if(WIN32 AND NOT BUILD_SHARED_LIBS)
+    target_link_libraries(simple_client PRIVATE crypt32.lib)
+    target_link_libraries(custom_storage PRIVATE crypt32.lib)
+    target_link_libraries(dump_torrent PRIVATE crypt32.lib)
+    target_link_libraries(make_torrent PRIVATE crypt32.lib)
+    target_link_libraries(connection_tester PRIVATE crypt32.lib)
+    target_link_libraries(upnp_test PRIVATE crypt32.lib)
+    target_link_libraries(client_test PRIVATE crypt32.lib)
```

Comment:
`crypt32.lib` looks like a dependency libtorrent has. I imagine this dependency still boils down to a dependency on a shared library (say, `crypt32.dll`), and this is the link library. And the reason you need this when building libtorrent as a static library is because static libraries cannot propagate dependencies on other shared libraries, the way a shared library can.

However, it seems more appropriate to propagate this dependency via the build system instead of like this (where the dependency is not propagated at all).

Couldn't you make the `libtorrent` target have a `PUBLIC` dependency on `crypt32.lib` or `crypt32.dll`, and have everyone linking against it pick that up?

Reply:
Although I do not use Windows, CMake should handle the ".lib" suffix automatically, because it handles the "lib" prefix on POSIX platforms.[Documentation](https://cmake.org/cmake/help/latest/command/target_link_libraries.html) seems to agree with that.

Reply:
Ah, yes: https://github.com/arvidn/libtorrent/blob/RC_1_2/CMakeLists.txt#L545
So, I imagine you just need to add something like `$<$<NOT:$<BOOL:BUILD_SHARED_LIBS>>:crypt32>`

Reply:
@FranciscoPombal could you give that a try?

Reply:
@arvidn yes, I should have that ready today

Reply:
@arvidn fixed.

Reply:
@zeule it's not clear to me why this dependency should not exist when building a shared library. I assume `crypt32` is a cmake target which refers to a DLL. A dependency libtorrent would have regardless of whether it's built as a static library or shared, no?

Reply:
For me too. Did not code for Windows for the last 10 years, have no idea what is libcrypto.lib in the linking error messages.